byzcoin/bcadmin/main.go

```diff
@@ -477,20 +493,24 @@ func link(c *cli.Context) error {
 		}
 		ad := &darc.Darc{}
 		adPub := cothority.Suite.Point()
-		if adIDStr := c.String("darc"); len(adIDStr) == 69 {
-			adID, err := hex.DecodeString(adIDStr[5:])
+		// Accept both plain-darcs, as well as "darc:...." darcs
```

Comment:
getDarcByString is already doing "strings.HasPrefix("darc:")" and then trimming it. Is there any way we can keep this in one place?

Reply:
OK, split `getDarcByString` to a string-to-id and a `getDarc` part. Also added `stringToEd25519Buf`.